app/questionnaire/path_finder.py

```diff
@@ -13,45 +18,48 @@
 
 
 class PathFinder:
-    def __init__(self, schema, answer_store, metadata, completed_blocks):
+    def __init__(
+        self,
+        schema: QuestionnaireSchema,
+        answer_store: AnswerStore,
+        metadata: Mapping,
+        completed: CompletedItems = None,
+    ):
         self.answer_store = answer_store
         self.metadata = metadata
         self.schema = schema
-        self.completed_blocks = completed_blocks
-        self._full_routing_path = None
+        self.completed_items = completed
 
-    @staticmethod
-    def _block_index_for_location(blocks, location):
-        return next(
-            (
-                index
-                for (index, block) in enumerate(blocks)
-                if block['id'] == location.block_id
-            ),
-            None,
-        )
+    def is_path_complete(self, path):
+        if self.get_first_incomplete_location_on_path(path):
```

Comment:
`return not self.get_first_incomplete_location_on_path(path)` ?

Reply:
Done